src/clang.rs

```diff
@@ -747,6 +741,36 @@ impl Type {
     }
 }
 
+/// An iterator for a type's template arguments
+pub struct TypeTemplateArgIterator {
+    x: CXType,
+    length: u32,
+    index: u32
+}
+
+impl Iterator for TypeTemplateArgIterator {
+    type Item = Type;
+    fn next(&mut self) -> Option<Type> {
+        if self.index < self.length {
+            let idx = self.index as c_int;
+            self.index += 1;
+            Some(Type { x: unsafe { clang_Type_getTemplateArgumentAsType(self.x, idx) } })
+        } else {
+            None
+        }
+    }
+}
+
+impl ExactSizeIterator for TypeTemplateArgIterator {
+    fn len(&self) -> usize {
+        if self.index < self.length {
```

Comment:
self.index can never be greater than self.length right? In that case, this if is not needed, since we'll always return the correct value.